### PR TITLE
Convert group to community

### PR DESCRIPTION
**Pull request by julianjelfs**
and a couple of other bits and bobs.

### CODE REVIEW

File: dfx.json [hunk 1]
```diff
@@ -1,5 +1,5 @@
 {
-  "dfx": "0.14.2",
+  "dfx": "0.13.1",
```
Comment (hpeebles) [inline dfx.json:2]: I think we should keep it on 0.14.2 but just for local testing you'll need to currently use 0.13.1.
When we deploy to prod we should still use latest.

File: frontend/app/src/components/home/CurrentChatMenu.svelte [hunk 2]
```diff
@@ -134,7 +137,9 @@
     }
 
     function convertToCommunity() {
-        dispatch("convertToCommunity", { chatId: selectedChatSummary.id });
+        if (selectedChatSummary.kind === "group_chat" && selectedChatSummary.public) {
```
Comment (hpeebles) [inline frontend/app/src/components/home/CurrentChatMenu.svelte:140]: You can convert private groups too. They simply become private communities.

File: frontend/app/src/i18n/en.json [hunk 3]
```diff
@@ -916,7 +921,8 @@
         "groupChats": "Group chats",
         "favourites": "Favourites",
         "addToFavourites": "Add to favourites",
-        "removeFromFavourites": "Remove from favourites"
+        "removeFromFavourites": "Remove from favourites",
+        "convertInfo": "We can take care of converting your public group to a community.\n\n This is an irreversible process and will do the following:\n\n* freeze the existing public group\n* create a new community with the same name as the group\n* add a default channel to the community with the same name as the group\n* add all members of the group as members of the community\n* copy the full message history from the group into the default channe\n* delete the existing group\n* inform all members of the group that it has been converted to a community\n\nThis process might take a few minutes depending on the size of the group so please be patient."
```
Comment (hpeebles) [inline frontend/app/src/i18n/en.json:925]: Should remove 'public'

Comment (julianjelfs) [reply frontend/app/src/i18n/en.json:925]: Should we? Can I convert a private group to a community?

File: frontend/openchat-agent/src/services/community/candid/idl.js [hunk 4]
```diff
@@ -372,6 +372,7 @@ export const idlFactory = ({ IDL }) => {
     'block_index' : BlockIndex,
     'from' : NnsCryptoAccount,
     'memo' : IDL.Nat64,
+    'ledger' : CanisterId,
```
Comment (hpeebles) [inline frontend/openchat-agent/src/services/community/candid/idl.js:375]: With these ledger fields added we won't be able to release the frontend until the backend goes out.
And we can't currently release the backend until we upgrade to candid 0.9.0 which we'll be able to do in a few days.

File: frontend/app/src/i18n/en.json (hunk 3 above)
Comment (hpeebles) [reply frontend/app/src/i18n/en.json:925]: Yup, it just becomes a private community.

File: frontend/openchat-agent/src/services/community/candid/idl.js (hunk 4 above)
Comment (hpeebles) [reply frontend/openchat-agent/src/services/community/candid/idl.js:375]: Which is maybe ok... but its something to be aware of.

File: frontend/openchat-agent/src/services/group/mappers.ts [hunk 5]
```diff
@@ -873,3 +876,16 @@ export function rulesResponse(candid: ApiRulesResponse): AccessRules | undefined
         };
     }
 }
+export function convertToCommunityReponse(
+    candid: ApiConvertIntoCommunityResponse
+): ConvertToCommunityResponse {
+    if ("Success" in candid) {
+        return {
+            kind: "success",
+            id: { kind: "community", communityId: candid.Success.toString() },
```
Comment (hpeebles) [inline frontend/openchat-agent/src/services/group/mappers.ts:885]: If you run codegen you'll get the channelId too.

File: dfx.json (hunk 1 above)
Comment (hpeebles) [reply dfx.json:2]: It's back!

**Review by hpeebles — APPROVED** [review]

**Pull request merged**